source/system/Item.h

```diff
@@ -62,91 +65,22 @@ struct ItemEffect
-		state(ResourceState::NotLoaded)
-	{
-	}
-	virtual ~Item() {}
-
-	Item& operator = (const Item& i);
-
-	template<typename T, ITEM_TYPE _type>
-	T& Cast()
-	{
-		assert(type == _type);
-		return *(T*)this;
-	}
-
-	template<typename T, ITEM_TYPE _type>
-	const T& Cast() const
-	{
-		assert(type == _type);
-		return *(const T*)this;
-	}
-
-	Weapon& ToWeapon() { return Cast<Weapon, IT_WEAPON>(); }
-	Bow& ToBow() { return Cast<Bow, IT_BOW>(); }
-	Shield& ToShield() { return Cast<Shield, IT_SHIELD>(); }
-	Armor& ToArmor() { return Cast<Armor, IT_ARMOR>(); }
-	Amulet& ToAmulet() { return Cast<Amulet, IT_AMULET>(); }
-	Ring& ToRing() { return Cast<Ring, IT_RING>(); }
-	Consumable& ToConsumable() { return Cast<Consumable, IT_CONSUMABLE>(); }
-	OtherItem& ToOther() { return Cast<OtherItem, IT_OTHER>(); }
-	Book& ToBook() { return Cast<Book, IT_BOOK>(); }
-
-	const Weapon& ToWeapon() const { return Cast<Weapon, IT_WEAPON>(); }
-	const Bow& ToBow() const { return Cast<Bow, IT_BOW>(); }
-	const Shield& ToShield() const { return Cast<Shield, IT_SHIELD>(); }
-	const Armor& ToArmor() const { return Cast<Armor, IT_ARMOR>(); }
-	const Amulet& ToAmulet() const { return Cast<Amulet, IT_AMULET>(); }
-	const Ring& ToRing() const { return Cast<Ring, IT_RING>(); }
-	const Consumable& ToConsumable() const { return Cast<Consumable, IT_CONSUMABLE>(); }
-	const OtherItem& ToOther() const { return Cast<OtherItem, IT_OTHER>(); }
-	const Book& ToBook() const { return Cast<Book, IT_BOOK>(); }
-
-	bool IsStackable() const { return Any(type, IT_CONSUMABLE, IT_GOLD, IT_BOOK) || (type == IT_OTHER && !IsSet(flags, ITEM_QUEST)); }
-	bool CanBeGenerated() const { return !IsSet(flags, ITEM_NOT_RANDOM); }
-	bool IsWearable() const { return Any(type, IT_WEAPON, IT_BOW, IT_SHIELD, IT_ARMOR, IT_AMULET, IT_RING); }
-	bool IsQuest() const { return IsSet(flags, ITEM_QUEST); }
-	bool IsQuest(int quest_id) const { return IsQuest() && this->quest_id == quest_id; }
-
-	const string& GetName() const { return name; }
-	float GetWeight() const { return float(weight) / 10; }
-	float GetWeightValue() const { return float(value) / weight; }
-	float GetEffectPower(EffectId effect) const;
-
-	void CreateCopy(Item& item) const;
-	Item* CreateCopy() const;
-	Item* QuestCopy(Quest* quest, const string& name);
-	void Rename(cstring name);
-	void RenameS(const string& name) { Rename(name.c_str()); }
-
-	string id, name, desc;
-	int weight, value, ai_value, flags, quest_id;
-	vector<ItemEffect> effects;
-	ITEM_TYPE type;
-	MeshPtr mesh;
-	TexturePtr tex, icon;
-	ResourceState state;
-
-	static const Item* gold;
-	static ItemsMap items;
-	static Item* TryGet(Cstring id);
-	static Item* Get(Cstring id)
-	{
-		auto item = TryGet(id);
-		assert(item);
-		return item;
-	}
-	static Item* GetS(const string& id)
-	{
-		return TryGet(id);
-	}
-	static void Validate(uint& err);
+protected:
+	friend Item;
+	Item* item;
```

Comment:
This for maintaining compatiblity with Item spawning and other mechanics, because original code estimates that it should have access to fields of what was originally base class and now it is a container. Usage of `item` field and its's getters is deprecated. New mechanics should not use this field / getter.